src/api/routes/transcripts.ts

```diff
@@ -52,8 +59,74 @@ const list = procedure
   return g.transcripts;
 });
 
+const getNameMap = procedure
+  .use(authUser())
+  .input(z.object({ transcriptId: z.string() }))
+  .output(z.record(z.string()))
+  .query(async ({ input }) => { return await extractAndGetNameMap(input.transcriptId); });
+
+// expected input should an object of {[handlebarNames]: userIds]}
+const updateNameMap = procedure
+  .use(authUser())
+  .input(z.record(z.string()))
+  .mutation(async ({ input: nameMap }) => 
+{
+  // Construct an array of objects to upsert multiple rows the same time
+  const upsertArray = Object.entries(nameMap)
+    .map(([handlebarName, userId]) => ({
+      handlebarName,
+      userId,
+    }));
+
+  await db.SummaryNameMapping.bulkCreate(upsertArray, { updateOnDuplicate: ['userId'] });
+});
 
 export default router({
   get,
   list,
+  getNameMap,
+  updateNameMap
 });
+
+export async function extractAndGetNameMap(transcriptId: string): Promise<Record<string, string>> {
```

Comment:
Please see the comment below. For the latest commit, I have assign the summary list to the transcript object as you suggested, but I think we could just use the returned `summaries` from this function directly instead of assigning it to the object then return.